beets/ui/commands.py

```diff
@@ -684,10 +704,18 @@ def choose_match(self, task):
         # Loop until we have a choice.
         candidates, rec = task.candidates, task.rec
         while True:
+            # Gather extra choices from plugins.
+            # TODO: check for conflicts in choices (duplicated letters, etc)
+            # TODO: provide a way to override summary_judment and/or non-timid
+            # mode? Might be useful for some plugins
+            extra_choices = list(chain(*plugins.send('before_choose_candidate',
```

Comment:
Yes, just crashing when there's a letter conflict would probably be the best policy.

I think we can defer other option-related functionality (like `summary_judgment`) to another iteration. It's a great point, but this will be fine for a first version of the hook.